repos/fdbt-site/src/pages/api/updateFareTriangle.ts

```diff
@@ -0,0 +1,342 @@
+import { NextApiResponse } from 'next';
+import uniq from 'lodash/uniq';
+import Papa from 'papaparse';
+import { putDataInS3 } from '../../data/s3';
+import { NextApiRequestWithSession, ErrorInfo, UserFareStages } from '../../interfaces';
+import { redirectToError, redirectTo } from '../../utils/apiUtils';
+import {
+    INPUT_METHOD_ATTRIBUTE,
+    CSV_UPLOAD_ATTRIBUTE,
+    MATCHING_JSON_ATTRIBUTE,
+    PRODUCT_AND_SERVICE_ID_ATTRIBUTE,
+} from '../../constants/attributes';
+import { getFormData, processFileUpload } from '../../utils/apiUtils/fileUpload';
+import logger from '../../utils/logger';
+import { getSessionAttribute, updateSessionAttribute } from '../../utils/sessions';
+import { WithIds, ReturnTicket, SingleTicket } from '../../../shared/matchingJsonTypes';
+
+const errorId = 'csv-upload';
+
+interface FareTriangleData {
+    fareStages: {
+        stageName: string;
+        prices: {
+            [key: string]: {
+                price: string;
+                fareZones: string[];
+            };
+        };
+    }[];
+}
+
+export const config = {
+    api: {
+        bodyParser: false,
+    },
+};
+
+export const setCsvUploadAttributeAndRedirect = (
+    req: NextApiRequestWithSession,
+    res: NextApiResponse,
+    errors: ErrorInfo[],
+    poundsOrPence?: string,
+): void => {
+    updateSessionAttribute(req, CSV_UPLOAD_ATTRIBUTE, { errors, poundsOrPence });
+
+    redirectTo(res, '/csvUpload');
+};
+
+export const containsDuplicateFareStages = (fareStageNames: string[]): boolean =>
```

Comment:
if we have some duplicated methods across both files, could we import the duplicated ones from the original file?

Reply:
Done, good shout.